### ISSUE TITLE
Rewrite Controls utils objects

### ISSUE THREAD

**Issue by w-4** [flarum/core#2414]
## Feature Request

Here's a screenshot of the before (left) and after:
![image](https://user-images.githubusercontent.com/36057469/97195939-764f0200-17de-11eb-8475-d7b6ebdf491f.png)
 
The biggest benefit is that when calling the function one doesn't have to use `.call(discussion)` anymore and the code becomes more readable. Another benefit is that you can call one utils function from another in the same file, which was previously impractical, since `this` was already bound to the discussion. 

Applies to `PostControls`, `DiscussionControls`, `UserControls`. Can be done probably when converting these files to TypeScript.

**Comment by datitisev** [comment]
> Can be done probably when converting these files to TypeScript.

Not entirely sure, but I think some of these actions are referenced from other files. The import statement would change and thus would be a breaking change.

Because right now we have the `import UserControls from '...'`, that'd have to be changed to `import * as UserControls from '...'` I think

**Comment by askvortsov1** [comment]
My biggest concern with this approach is that the `extend` and `override` utils wouldn't work on standalone functions.

**Comment by w-4** [comment]
That's true, we'd need to also export it as an object in addition. That's also needed for the extensions public API, so that webpack uglify does not alter the names. 
```js
export default DiscussionControls = {
  userControls,
  moderationControls,
  destructiveControls,
  ...
};
```

**Comment by datitisev** [comment]
It doesn't matter what uglify does, as the "public API" comes from the `compat` files, where we use strings and so they aren't affected by uglify. Extensibility _is_ an issue that could maybe be solved with that method, however.

**Comment by askvortsov1** [comment]
We should also consider some of the options discussed in https://github.com/flarum/core/issues/1317
